src/webgpu/shader/execution/expression/precedence.spec.ts

```diff
@@ -0,0 +1,96 @@
+export const description = `
+Execution tests for operator precedence.
+`;
+
+import { makeTestGroup } from '../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../common/util/data_tables.js';
+import { GPUTest } from '../../../gpu_test.js';
+
+export const g = makeTestGroup(GPUTest);
+
+// The list of test cases and their expected results.
+interface Expression {
+  expr: string;
+  result: number;
+}
+const kExpressions: Record<string, Expression> = {
```

Comment:
Suggestions:
* Parameterize the test to no-op or strip out all whitespace.
* Extend the `decl` parameterization to replace the variable names with their literal value (`three` -> `3`).

Reply:
Done.